File: 1603.设计停车系统.go
/*
 * @lc app=leetcode.cn id=1603 lang=golang
 *
 * [1603] 设计停车系统
 */

// @lc code=start
type ParkingSystem struct {
	Big        int
	Medium     int
	Small      int
	CurrBig    int
	currMedium int
	currSmall  int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	// 其余默认为 0 即可
	return ParkingSystem{
		Big:    big,
		Medium: medium,
		Small:  small,
	}
}

func (this *ParkingSystem) AddCar(carType int) bool {
	if carType == 1 && this.CurrBig < this.Big {
		this.CurrBig++
	} else if carType == 2 && this.currMedium < this.Medium {
		this.currMedium++
	} else if carType == 3 && this.currSmall < this.Small {
		this.currSmall++
	} else {
		return false
	}
	return true
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */
// @lc code=end

File: 1603.设计停车系统.py
#
# @lc app=leetcode.cn id=1603 lang=python3
#
# [1603] 设计停车系统
#

# @lc code=start
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.big = big
        self.medium = medium
        self.small = small
        self.curr_big = 0
        self.curr_medium = 0
        self.curr_small = 0


    def addCar(self, carType: int) -> bool:
        if carType == 1 and self.curr_big < self.big:
            self.curr_big += 1
        elif carType == 2 and self.curr_medium < self.medium:
            self.curr_medium += 1
        elif carType == 3 and self.curr_small < self.small:
            self.curr_small += 1
        else:
            # 没有空余车位
            return False
        return True



# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)
# @lc code=end

